Automattic/pocket-casts-android
# Implement Large Player widget

## Description

This targets a separate branch instead of `main` because widgets cannot be feature flagged as they are compile time dependent. Once all features are implement I will make a PR for the `task/widgets` branch.

Designs: Lkt7fuf9Nq3XvfAFPCfpTD-fi-1044_7170

There are some known issues:

1. Sometimes artwork doesn't load and placeholder is displayed instead. Even though the `Image` composable receives a correct `ImageProvider` it still doesn't update. It might be a bug withing the `Glance` library. However image loads correctly once the playback is resumed.
2. Widget's preview and placeholder are currently not implemented. Internal ref: p1711476556931619/1711396714.667229-slack-C05RR9P9RAT

This PR also improves accessibility by using better colors for buttons.

## Testing Instructions

There are many things to test with this widget and it's hard to provide all instructions. Do some monkey tests on the widget.
1. Play episodes from the queue.
2. Change queue in the app and see if the widget updates.
3. Control playback.
4. See if time left updates (it updates in 5 second increments).

## Screenshots or Screencast 

### Demo

https://github.com/Automattic/pocket-casts-android/assets/30936061/500005aa-9a1d-4c7b-bc2e-d8910027dfcb

### Dark and light mod

| Dark | Light |
| - | - |
| ![Screenshot_20240328-143325](https://github.com/Automattic/pocket-casts-android/assets/30936061/9b1fd62a-e546-478b-afe9-caeef973e17d) | ![Screenshot_20240328-143334](https://github.com/Automattic/pocket-casts-android/assets/30936061/a8d70066-1a5e-4511-9f86-3fef2a9f4c7c) |

### Dynamic themes

| A | B | C | D |
| - | - | - | - |
| ![Screenshot_20240328-143143](https://github.com/Automattic/pocket-casts-android/assets/30936061/c547542c-f785-4920-8ac7-b6808e8bda27) | ![Screenshot_20240328-143152](https://github.com/Automattic/pocket-casts-android/assets/30936061/5b7733e7-6248-42a0-a865-d57d12b011e3) | ![Screenshot_20240328-143205](https://github.com/Automattic/pocket-casts-android/assets/30936061/3f767943-3e5a-404d-a1fe-1c4e4fb1d94f) | ![Screenshot_20240328-143308](https://github.com/Automattic/pocket-casts-android/assets/30936061/7b83562e-fed5-4103-a001-357cebc9d2cb) |

## Checklist
 
#### I have tested any UI changes...
